x/precisebank/types/fractional_balance.go

```diff
@@ -39,5 +41,24 @@ func (fb FractionalBalance) Validate() error {
 	}
 
 	// Validate the amount with the FractionalAmount wrapper
-	return NewFractionalAmountFromInt(fb.Amount).Validate()
+	return ValidateFractionalAmount(fb.Amount)
+}
+
+// ValidateFractionalAmount checks if an sdkmath.Int is a valid fractional
+// amount, ensuring it is positive and less than or equal to the maximum
+// fractional amount.
+func ValidateFractionalAmount(amt sdkmath.Int) error {
+	if amt.IsNil() {
+		return fmt.Errorf("nil amount")
+	}
+
+	if !amt.IsPositive() {
+		return fmt.Errorf("non-positive amount %v", amt)
+	}
+
+	if amt.GT(maxFractionalAmount) {
```

Comment:
Should this be `GTE` - ie the fraction should always be less than 1ukava, or is there a case where it's exactly 1ukava

Reply:
maxFractionalAmount would be (1_000_000_000_000 - 1)
though removed the use of `maxFractionalAmount` now and replacing it with `conversionFactor` directly now, which also matches this change